extensions/amp-a4a/0.1/refresh-manager.js

```diff
@@ -0,0 +1,202 @@
+   * element.
+   *
+   * @return {?Promise} Promise that resolves after refresh event. This promise
+   *   is useful to have for test purposes.
+   */
+  initiateRefreshCycle() {
+    if (!this.isRefreshable()) {
+      return null;
+    }
+    return new Promise(resolve => {
+      analyticsForDoc(this.element_, true).then(analytics => {
+        analytics.getAnalyticsRoot(this.element_).getVisibilityManager()
+            .listenElement(this.element_, this.config_, null, null, () => {
+              this.refreshTimeoutId_ = this.timer_.delay(() => {
+                this.a4a_.refresh(() => this.initiateRefreshCycle());
+                resolve();
+              }, /** @type {number} */ (this.refreshInterval_));
+            });
+      });
+    });
+  }
+
+  /**
+   * Terminates the current refresh cycle, if one currently exists. Returns
+   * true if the cycle was canceled successfully, false otherwise.
+   */
+  stopRefreshCycle() {
+    if (this.refreshTimeoutId_) {
+      this.timer_.cancel(this.refreshTimeoutId_);
+      this.refreshTimeoutId_ = null;
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Converts config to appropriate units.
+   * @return {!RefreshConfig}
+   */
+  getConfiguration_(config) {
+    // Convert seconds to milliseconds.
+    config['totalTimeMin'] *= 1000;
+    config['continuousTimeMin'] *= 1000;
+    return config;
+  }
+
+  /**
+   * Retrieves the publisher-specified refresh interval, if one were set.
+   *
+   * @return {?number}
+   */
+  getPublisherSpecifiedRefreshInterval_() {
+    const refreshInterval = this.element_.getAttribute(DATA_ATTR_NAME);
+    if (refreshInterval) {
+      return this.checkAndSanitizeRefreshInterval_(refreshInterval);
+    }
+    let metaTag;
+    const metaTagContent = ((metaTag = this.win_.document
+          .getElementsByName(METATAG_NAME))
+        && metaTag[0]
+        && metaTag[0].getAttribute('content'));
+    const networkIntervalPairs =
+        metaTagContent ? metaTagContent.split(',') : [];
+    for (let i = 0; i < networkIntervalPairs.length; i++) {
+      const pair = networkIntervalPairs[i].split('=');
+      if (pair.length != 2) {
+        user().warn(TAG,
+            'refresh metadata config must be of the form ' +
+            '`network_type=refresh_interval`');
+      } else if (pair[0] == this.adType_) {
+        return this.checkAndSanitizeRefreshInterval_(pair[1]);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Ensures that refreshInterval is a number no less than 30. Returns null if
+   * the given input fails to meet these criteria. This also converts from
+   * seconds to milliseconds.
+   *
+   * @param {(number|string)} refreshInterval
+   * @return {?number}
+   */
+  checkAndSanitizeRefreshInterval_(refreshInterval) {
+    if (isNaN(refreshInterval) || refreshInterval == '') {
+      user().warn(TAG, 'refresh interval must be a number');
+      return null;
+    } else if (refreshInterval < MIN_REFRESH_INTERVAL) {
```

Comment:
I don't believe you are properly handling a string based interval here.  Perhaps it should be:
const refreshIntervalNum = Number(refreshInterval);
dev().assert(!isNaN(refreshInterval) && refreshInterval >= MIN_REFRESH_INTERVAL, 'invalid refresh interval, must be at least ${MIN_REFRESH_INTERVAL}s: ${refreshInterval}`);